projects/coordinator/src/main/java/org/batfish/coordinator/resources/ContainerResource.java

```diff
@@ -0,0 +1,110 @@
+package org.batfish.coordinator.resources;
+
+import java.nio.file.Files;
+import java.util.SortedSet;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.ForbiddenException;
+import javax.ws.rs.GET;
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import javax.ws.rs.core.UriBuilder;
+import javax.ws.rs.core.UriInfo;
+import org.batfish.common.BatfishLogger;
+import org.batfish.common.Container;
+import org.batfish.common.CoordConsts;
+import org.batfish.coordinator.Main;
+
+/**
+ * The ContainerResource is a RESTful service for servicing client API calls at container level.
+ *
+ * <p>The ContainerResource provides information about a specified container, and provides the
+ * ability to create/delete a specified container for authenticated clients.
+ *
+ * <p>The ContainerResource also provides the access entry for other client API calls.
+ */
+@Produces(MediaType.APPLICATION_JSON)
+public class ContainerResource {
+
+  private BatfishLogger _logger = Main.getLogger();
+
+  private UriInfo _uriInfo;
+  private String _name;
+  private String _apiKey;
+
+  public ContainerResource(UriInfo uriInfo, String apiKey, String name) {
+    _uriInfo = uriInfo;
+    _apiKey = apiKey;
+    _name = name;
+  }
+
+  /** Returns information about the given {@link Container}, provided this user can access it. */
+  @GET
+  public Response getContainer() {
+    _logger.info("WMS: getContainer '" + _name + "'\n");
+    validate();
+    Container container = Main.getWorkMgr().getContainer(_name);
+    return Response.ok(container).build();
+  }
+
+  /** Initialize a new container with name: {@link #_name}. */
+  @POST
+  public Response initContainer() {
```

Comment:
This relates back to my previous comment:
- I think there should be a `@POST` handler for the `/container` URL itself, and the body should be something with multiple optional fields: `name`, `prefix`, etc. That is to say, the `name` should not be in the URL for create -- instead, it should be a name or prefix request.



Aside, to match more what it's doing in an API sense, we should call the function createContainer, not initContainer.